ferry.go

```diff
@@ -890,6 +908,21 @@ func (f *Ferry) ReportProgress() {
 	}
 }
 
+func (f *Ferry) ReportState() {
+	callback := f.Config.StateCallback
+	state, err := f.SerializeStateToJSON()
+	if err != nil {
+		f.logger.WithError(err).Error("failed to dump state to JSON")
+		return
+	}
+
+	callback.Payload = string(state)
+	err = callback.Post(&http.Client{})
+	if err != nil {
+		f.logger.WithError(err).Errorf("failed to post state to callback: %s", callback)
```

Comment:
I made this an `Error` instead of `Warn` (like above) considering the importance and how it will be used versus simply a progress report.

Reply:
I don't think the `.Error` method actually does anything other than emit to stderr. Maybe you want to panic via the `ErrorHandler`?

Reply:
Same comment below. I did want to raise it in the logs as an error instead of just a warning. If we do want this to panic and fail the run we can, although it isn't really necessary. Again, if you feel differently I could be convinced.